src/XrdTls/XrdTlsTempCA.cc

```diff
@@ -198,11 +209,17 @@ CRLSet::processFile(file_smart_ptr &fp, const std::string &fname)
             return false;
         }
     }
+    if(!m_atLeastOneValidCRLFound)
+        m_atLeastOneValidCRLFound = atLeastOneValidCRLFound;
```

Comment:
Doesn't this mean that once a valid CRL is found a subsequent scan that doesn't find one will not be registered as such? Won't that be a problem?

Reply:
I only modify the variable `m_atLeastOneValidCRLFound` if this variable is false. So, once a valid CRL is found, the `m_atLeastOneValidCRLFound` will be set to true and therefore will never be assigned anymore during the `XrdTlsTempCA::Maintenance()` loop.

That way, we only enable CRL checking if there is at least one valid CRL found during the Maintenance run :)

Reply:
What about this little issue that doesn't seem quite right.

Reply:
Yes, but you should consider that the next time the maintenance loop runs there may be no valid CRLs. I know that's unlikely but unlikely things happen. I think you should always update the current state of the maintenance loop otherwise that look does half the j0ob.

Reply:
If during the next time the maintenance loop runs there is no valid CRL file, do we agree that the CRL checking must be disabled?

Reply:
Ok, I see what you mean. I just need to reset this boolean to false at the beginning of the Maintenance run :+1:

Reply:
I think that if the current CRL file is replaced with a bad CRL file then, yes. It should be disabled. This is dynamic because the maintenance check tries to create a CRL file and if it fails then a) if the previous CRL file is still enforce then we need do nothing, or b) it gets replaced with a null file then we need top disable CRL checking. So, in the end it depends on the state the maintenance run leaves the CRL file (something I have not researched).